python/hidet/graph/ops/conv2d/conv2d.py

```diff
@@ -39,15 +51,24 @@ def __init__(self, data: TensorNode, weight: TensorNode, stride: List[int], dila
             ),
         )
         out_group_size = oc // groups
+        # upcast to the resulting dtype
+        dtype = data.type.dtype if data.type.dtype.nbytes > weight.type.dtype.nbytes else weight.type.dtype
+
+        def f_compute(wci, kxi, kyi, ni, oci, pi, qi):
+            im_hi = pi * sx + kxi * dilx
+            im_wi = qi * sy + kyi * dily
+            return if_then_else(
+                logical_and(im_hi >= pad_h, im_hi < h + pad_h, im_wi >= pad_w, im_wi < w + pad_w),
+                data[ni, (oci // out_group_size) * wc + wci, im_hi - pad_h, im_wi - pad_w] * weight[oci, wci, kxi, kyi],
+                dtype(pad_value),
+            )
```

Comment:
We can define a group of prelimiary computation definitions and put it at the "compute ops" (https://github.com/hidet-org/hidet/tree/main/python/hidet/ir/compute/cops) submodule. 

In this case, we can create a submodule hidet.ir.compute.cops.pad to store the computation for padding, and replace the padding computation in pad op and conv2d op.

In the future, when we write a specialized implementation for some operator (like matmul or conv2d), we can reuse the computation definitions.